### ISSUE TITLE
Support /users endpoints in API v2

### ISSUE THREAD

**Issue by albullington** [inaturalist/iNaturalistAPI#281]
For iNatRN, we need to be able to view the current user profile, as well as other user's profiles via API v2. We also need to be able to edit the current user's profile, and fetch projects the current user has joined or followed.

We'll need a way to see the user's dashboard, similar to the dashboard redesign spec, with the people I follow, places/taxa I follow, and projects I follow categories.

- [x] GET /users/me
- [x] GET /users/:id
- [x] PUT /users/:id
- [x] GET /users/:id/projects
- [x] GET /users/dashboard_updates

It's likely that we'll also need to be able to mute/unmute and block/unblock users via a settings screen (similar to user settings on the web), but we can make a separate issue for this later when the MVP flows are more finalized.

**Comment by pleary** [comment]
The endpoints listed here should be working. Instead of `users/dashboard_updates` I brought over `observations/updates` which is what at least the  iOS app is using now for updates. If we need something different it would be good to get that spec'd out
